presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetHiveRecordCursor.java

```diff
@@ -735,7 +743,7 @@ public ParquetStructConverter(Type prestoType, String columnName, GroupType entr
             List<Type> prestoTypeParameters = prestoType.getTypeParameters();
             List<parquet.schema.Type> fieldTypes = entryType.getFields();
             checkArgument(
-                    prestoTypeParameters.size() == fieldTypes.size(),
```

Comment:
This check is good but too restrictive. It makes it hard to evolve the schema seamlessly. We return nulls and log some message below so that queries can still go on. If they pay attention to the logging, they should realize what's happened.

Even if the number matches, the schema still could be incompatible. Users still need to know what they are doing.